draco/util/gaussian_process.py

```diff
@@ -0,0 +1,353 @@
+                # to `interp_dtype`
+                tmp[:, kk] = _fast_tools._matmul_banded(
+                    A, xij[kk].astype(np.float64), start, end
+                )
+            # Write to the output data array with the original data
+            # dtype (depending on whether it's real or complex)
+            xout[ii, mt, jj] = tmp.view(data_dtype)[:, 0]
+
+    # Invert the variance to get weights. Faster to do one
+    # operation here instead of inverting in the for loop
+    invert_no_zero(wout, out=wout)
+
+    # Weights shouldn't be negative - this is probably
+    # numerical error in a small number of samples
+    wout[wout < 0] = 0.0
+    xout[wout < 0] = 0.0
+
+    return xout, wout
+
+
+def _select_interp_samples(
+    xi: np.ndarray[np.number],
+    xo: np.ndarray[np.number],
+    mask: np.ndarray[bool],
+    kwidth: int,
+    cutoff: float,
+    partition: int = 0,
+) -> np.ndarray[bool]:
+    """Mask output samples which are more than some distance from input samples.
+
+    Parameters
+    ----------
+    xi
+        Samples where data points have been measured
+    xo
+        Samples we want to interpolate onto
+    mask
+        Mask corresponding to measured samples. `True` values are assumed
+        to be unflagged.
+    kwidth
+        Width of the interpolating kernel. Select only samples
+    cutoff
+        Select only output samples closer (in number of _input_ samples)
+        than this value to the `n`th nearest unflagged input sample.
+    partition
+        `n`th closest sample to consider when applying cutoff. This is fed into
+        `np.partition`. A value of `0` is relative to the nearest sample. Default
+        is 0.
+
+    Returns
+    -------
+    out
+        Mask with flagged samples set to `False`
+    """
+    dist = np.subtract.outer(xo, xi)
+    # Divide by the sample width to get the distance in number of input samples
+    dist /= np.median(np.abs(np.diff(xi)))
+
+    out = np.empty((mask.shape[0], xo.shape[0]), dtype=bool)
+
+    kw_cutoff = kwidth - 1
+
+    # Iterate over the first axis of `mask`
+    for ii in range(mask.shape[0]):
+        mi = mask[ii]
+
+        if not np.any(mi):
+            out[ii] = False
+            continue
+
+        dmi = dist[:, mi]
+
+        pdist = np.min(dmi, axis=-1, where=dmi > 0, initial=kw_cutoff)
+        ndist = np.max(dmi, axis=-1, where=dmi < 0, initial=-kw_cutoff)
+
+        out[ii] = np.maximum(pdist, abs(ndist)) < kw_cutoff
+        out[ii] &= np.partition(abs(dmi), partition, axis=-1)[:, partition] < cutoff
+
+    return out
+
+
+def _combine_gp_kernels_from_specs(
+    samples: tuple, kernel_params: list[dict] | tuple[dict] | dict
+) -> tuple[np.ndarray[np.floating], np.ndarray[np.floating]] | tuple[None, None]:
+    """Helper to combine multiple kernels from different specs."""
+    kernel_params = (
+        [kernel_params]
+        if not isinstance(kernel_params, list | tuple)
+        else kernel_params
+    )
```

Comment:
```suggestion
    if not isinstance(kernel_params, list | tuple):
        kernel_params = [kernel_params]
```
no need to assign something to itself